drivers/SerialBase.h

```diff
@@ -38,8 +38,8 @@ namespace mbed {
  * @{
  */
 
-/** A base class for serial port implementations
- * Can't be instantiated directly (use Serial or RawSerial)
+/** An abstract base class for serial port implementations
+ * Can't be instantiated in sub-classess
```

Comment:
"Abstract classes are useful when creating hierarchies of classes that model reality because they make it possible to **specify an invariant level of functionality in some methods, but leave the implementation of other methods** until a specific implementation of that class (a derived class) is needed."
Source: https://www.techopedia.com/definition/17408/abstract-class

An abstract class does not mean it has pure virtual functions

By the above, `SerialBase` is an abstract class.

Reply:
This is not true. Even in the link you provide it is said later on "An abstract class has at least one abstract method. An abstract method will not have any code in the base class; the code will be added in its derived classes."
See also https://en.cppreference.com/w/cpp/language/abstract_class : `An abstract class is a class that either defines or inherits at least one function for which the final overrider is pure virtual.`

Reply:
[Done](https://github.com/hugueskamba/mbed-os/compare/a320f0074eba213a9eac6e4448d74fae1d7e3439..d093760d60feea9db5bf7f67ad70257db3766fd0).